### DISCUSSION TITLE
Pyside6 and snake case with ui file

### DISCUSSION THREAD

**Discussion by paolodestefani** [pyinstaller/pyinstaller#7440] [Help] [answered]
Hi all
So i created a small program that uses snake_case and an external file created with QtDesigner. The .ui file is converted in a python file that is imported and used in the main file. Running this small program from VSCode or terminal works like a charme. I packed this small program with pyinstaller and i get the errors shown.
The problem is in this line of the log.txt file:

AttributeError: 'MainWindow' object has no attribute 'objectName'. Did you mean: 'object_name'? 

The imported module should not use snake_case. 

So what could be the problem ?
[log.txt](https://github.com/pyinstaller/pyinstaller/files/10716316/log.txt)
[HelloWorld.zip](https://github.com/pyinstaller/pyinstaller/files/10716330/HelloWorld.zip)

**Comment by rokm** [comment, marked as answer]
Hmm, looks like in unfrozen version, `snake_case` and `true_property` are applied only within the entry-point script (HelloWorld.py), i.e., `dir(self)` within `MainWindow`'s constructor shows snake_case attributes. Trying to access `objectName` there also results in the error. However, within the MainWindow.py, these are not applied (running `dir` on the passed `MainWindow` object in `setupUi` shows the camelCase attributes).

Whereas in the frozen application, snake_case attributes are applied in both modules. So whatever logic PySide6 is using to scope these features to individual modules seems to be failing within the frozen application for whatever reason. Will dig around a bit to see if I can figure what's going on, and if it can be fixed.

**Reply by rokm** [reply]
So the bad news is that `PySide6` uses modules' source code to determine whether they use `PySide6` or not ([here](https://github.com/qtproject/pyside-pyside-setup/blob/67f2a02dbbec765d69c1a43031e75b9eb0118a4d/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/feature.py#L140-L161), called from [here for already-loaded modules](https://github.com/qtproject/pyside-pyside-setup/blob/67f2a02dbbec765d69c1a43031e75b9eb0118a4d/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/feature.py#L123) and [here for newly loaded ones](https://github.com/qtproject/pyside-pyside-setup/blob/67f2a02dbbec765d69c1a43031e75b9eb0118a4d/sources/shiboken6/shibokenmodule/files.dir/shibokensupport/feature.py#L137)), and this information is used to reset the flags on per-module basis.

As we do not collect sources by default, all modules end up with their flag set to -1 (= ignore), and so the entry-point's setting 3 (= snake case, true properties) is used everywhere.

To be honest, I fail to see a practical use for having a codebase with mixed snake_case/true_properties settings. 

But if you really want to go down that way and freeze your application, you will have to either 

A) ensure that the source code for your modules is also collected.

E.g., `pyinstaller --clean --noconfirm HelloWorld.py --add-data "Ui/MainWindow.py:Ui"` for this particular example. For anything more complex, it might be better to use [package/module collection mode setting](https://pyinstaller.org/en/stable/hooks.html#package-collection-mode) for your `Ui` package (the `module_collection_mode` dictionary can be set either in a hook as described in that link, or in the .spec file, by passing optional `module_collection_mode` argument to `Analysis`).

B) "cheat" and manually set the flags value in PySide's features dictionary for modules where you want the features to be reset (because you know that they use `PySide6` and do not have the features turned on).

I.e., add the following block to your entry-point script, after the imports:

```python
# Must be done after importing PySide6
if getattr(sys, 'frozen', False): 
    import shibokensupport.feature
    shibokensupport.feature.pyside_feature_dict['Ui.MainWindow'] = 0
```

**Reply by paolodestefani** [reply]
OK
I tried the A) option and works so i will use this
I think PySide6 developers didn't port the ui tools to the snake_case feature yet so QtDesigner and pyside-uic still generate only camelcase python code from ui files. That's why the snake_case feater can be set per module. But maybe in future...

Thanks. I'm impressed how clear and precise are you answers!
